cmd/lambda/publish_lease_events/main.go

```diff
@@ -94,38 +94,32 @@ func handleRecord(input *handleRecordInput) error {
 		prevLeaseStatus := prevLeaseStatusAttr.String()
 
 		// Grab the new lease status
-		nextLeaseStatusAttr, ok := record.Change.NewImage["LeaseStatus"]
+		newLeaseStatusAttr, ok := record.Change.NewImage["LeaseStatus"]
 		if !ok {
 			return errors.New("next lease missing LeaseStatus attr")
 		}
-		nextLeaseStatus := nextLeaseStatusAttr.String()
+		newLeaseStatus := newLeaseStatusAttr.String()
 
-		if prevLeaseStatus == nextLeaseStatus {
+		if prevLeaseStatus == newLeaseStatus {
 			log.Print("Lease status has not changed.")
 			return nil
 		}
 
-		log.Printf("Transitioning from %s to %s", prevLeaseStatus, nextLeaseStatus)
+		log.Printf("Lease status changed from %s to %s", prevLeaseStatus, newLeaseStatus)
 
 		// Lease is now expired if it transitioned from "Active" --> "Inactive"
-		didBecomeInactive := isActiveStatus(prevLeaseStatus) && !isActiveStatus(nextLeaseStatus)
+		didBecomeInactive := isActiveStatus(prevLeaseStatus) && !isActiveStatus(newLeaseStatus)
 
 		if didBecomeInactive {
-			// Before adding the account to any queues, make sure the account is
-			// updated to NotReady state.
-			var acct *db.Account
-			acct, err := input.dbSvc.TransitionAccountStatus(lease.AccountID, db.Leased, db.NotReady)
+			// Set account status to NotReady
+			acct, err := input.dbSvc.GetAccount(lease.AccountID)
 			if err != nil {
-				log.Printf("ERROR: Failed to mark AccountStatus=NotReady for %s, after lease for %s became inactive",
-					lease.AccountID, lease.PrincipalID)
+				return fmt.Errorf("ERROR: Failed to retrieve account %s: %s", lease.AccountID, err)
```

Comment:
we took this out and back in.  The concern here is how dynamodb streams and Lambdas work.  Anything that can cause the Lambda to error will result in re-attempted processing of that message until its clear.  These default values are large.

Also can you change this to errors.InternalServerError.  Those errors don't just have to be used for http status errors that's just a side benefit.

Reply:
Yep, good point. Let's talk through the situation here in more detail. I think there's a better solution to be had.